_episodes/16-parallel.md

```diff
@@ -25,632 +23,430 @@ We now have the tools we need to run a multi-processor job. This is a very
-with the number of samples used.
-There could be some variation in additional runs of the script with the same
-number of samples, since the elapsed time is affected by other programs
-running on the computer at the same time.
-But if the script is the most computationally-intensive process running at the
-time, its calculations are the largest influence on the elapsed time.
-
-Now that we've developed our initial script to estimate π, we can see
-that as we increase the number of samples:
-
-1. The estimate of π tends to become more accurate.
-1. The amount of memory required scales approximately linearly.
-1. The amount of time to calculate scales approximately linearly.
-
-In general, achieving a better estimate of π requires a greater number of
-points.
-Take a closer look at `inside_circle`: should we expect to get high accuracy
-on a single machine?
-
-Probably not.
-The function allocates three arrays of size _N_ equal to the number of points
-belonging to this process.
-Using 64-bit floating point numbers, the memory footprint of these arrays can
-get quite large.
-Each 100,000,000 points sampled consumes 2.24 GiB of memory.
-Sampling 400,000,000 points consumes 8.94 GiB of memory,
-and if your machine has less RAM than that, it will grind to a halt.
-If you have 16 GiB installed, you won't quite make it to 750,000,000 points.
-
-## Running the Serial Job on a Compute Node
+{: .language-bash}
 
-Create a submission file, requesting one task on a single node and enough
-memory to prevent the job from running out of memory:
+{% include {{ site.snippets }}/parallel/one-task-jobscript.snip %}
 
 ```
-{{ site.remote.prompt }} nano serial-pi.sh
-{{ site.remote.prompt }} cat serial-pi.sh
+{{ site.remote.prompt }} {{ site.sched.submit.name }} serial-job.sh
 ```
 {: .language-bash}
 
-{% include {{ site.snippets }}/parallel/one-task-with-memory-jobscript.snip %}
-
-Then submit your job. We will use the batch file to set the options,
-rather than the command line.
+As before, use the {{ site.sched.name }} status commands to check whether your job
+is running and when it ends:
 
 ```
-{{ site.remote.prompt }} {{ site.sched.submit.name }} serial-pi.sh
+{{ site.remote.prompt }} {{ site.sched.status }} {{ site.sched.flag.user }}
 ```
 {: .language-bash}
 
-As before, use the status commands to check when your job runs.
-Use `ls` to locate the output file, and examine it. Is it what you expected?
-
-* How good is the value for π?
-* How much memory did it need?
-* How long did the job take to run?
+Use `ls` to locate the output file. The `-t` flag sorts in
+reverse-chronological order: newest first. What was the output?
 
-Modify the job script to increase both the number of samples and the amount
-of memory requested (perhaps by a factor of 2, then by a factor of 10),
-and resubmit the job each time.
+> ## Read the Job Output
+>
+> The cluster output should be written to a file in the folder you launched the
```

Comment:
Is this true for all schedulers? They should be able to figure out where their output files are from previous content, so perhaps we should jump straight into looking at contents of the output rather than get caught up in scheduler/site specifics

Reply:
Perhaps rephrase and use something like "on many clusters, this is the folder you launched the job from, but your instructor will provide guidance"

Reply:
I don't know for certain, but I certainly hope this applies to all schedulers running on Linux clusters. The default behavior is to use the current directory for I/O, unless the submission script specifies something different, which we have not done here. If we make the lesson too generic, it loses value.